libeos-updater-util/avahi-service-file.h

```diff
@@ -44,4 +46,160 @@ gboolean eos_avahi_service_file_delete (const gchar   *avahi_service_directory,
+ *
+ * - summary timestamp,
+ *   - key: "st",
+ *   - type: "t" (uint64)
+ *   - contents: a unix utc timestamp of the summary, ideally telling when
+ *               the original summary was created, otherwise it could also
+ *               be the modification time of the summary file on host
+ */
+
+/**
+ * EOS_OSTREE_AVAHI_OPTION_FORCE_VERSION_Y:
+ *
+ * TODO
+ */
+#define EOS_OSTREE_AVAHI_OPTION_FORCE_VERSION_Y "force-version"
+/**
+ * EOS_OSTREE_AVAHI_OPTION_BLOOM_HASH_ID_Y:
+ *
+ * TODO
+ */
+#define EOS_OSTREE_AVAHI_OPTION_BLOOM_HASH_ID_Y "bloom-hash-id"
+/**
+ * EOS_OSTREE_AVAHI_OPTION_BLOOM_K_Y:
+ *
+ * TODO
+ */
+#define EOS_OSTREE_AVAHI_OPTION_BLOOM_K_Y "bloom-k"
+/**
+ * EOS_OSTREE_AVAHI_OPTION_BLOOM_SIZE_U:
+ *
+ * TODO
+ */
+#define EOS_OSTREE_AVAHI_OPTION_BLOOM_SIZE_U "bloom-size"
+/**
+ * EOS_OSTREE_AVAHI_OPTION_REPO_INDEX_Q:
+ *
+ * TODO
+ */
+#define EOS_OSTREE_AVAHI_OPTION_REPO_INDEX_Q "repository-index"
+/**
+ * EOS_OSTREE_AVAHI_OPTION_PORT_Q:
+ *
+ * TODO
+ */
+#define EOS_OSTREE_AVAHI_OPTION_PORT_Q "port"
+/**
+ * EOS_OSTREE_AVAHI_OPTION_TXT_RECORDS_SIZE_LEVEL_Y:
+ *
+ * TODO
+ */
+#define EOS_OSTREE_AVAHI_OPTION_TXT_RECORDS_SIZE_LEVEL_Y "txt-records-size-level"
+/**
+ * EOS_OSTREE_AVAHI_OPTION_TXT_RECORDS_CUSTOM_SIZE_T:
+ *
+ * TODO
+ */
+#define EOS_OSTREE_AVAHI_OPTION_TXT_RECORDS_CUSTOM_SIZE_T "txt-records-custom-size"
+
+/* values for the EOS_OSTREE_AVAHI_OPTION_BLOOM_HASH_ID_Y option */
+
+/**
+ * EOS_OSTREE_AVAHI_STR_BLOOM_HASH_ID:
+ *
+ * TODO
+ */
+#define EOS_OSTREE_AVAHI_STR_BLOOM_HASH_ID ((guchar)1)
+
+/* values for the EOS_OSTREE_AVAHI_OPTION_TXT_RECORDS_SIZE_LEVEL_Y option */
+
+/**
+ * EOS_OSTREE_AVAHI_SIZE_LEVEL_CUSTOM:
+ *
+ * TODO
+ */
+#define EOS_OSTREE_AVAHI_SIZE_LEVEL_CUSTOM ((guchar)0)
+/**
+ * EOS_OSTREE_AVAHI_SIZE_LEVEL_SUPPORT_CRAPPY_HARDWARE:
+ *
+ * TODO
+ */
+#define EOS_OSTREE_AVAHI_SIZE_LEVEL_SUPPORT_CRAPPY_HARDWARE ((guchar)1)
+/**
+ * EOS_OSTREE_AVAHI_SIZE_LEVEL_FIT_SINGLE_DNS_MESSAGE:
+ *
+ * TODO
+ */
+#define EOS_OSTREE_AVAHI_SIZE_LEVEL_FIT_SINGLE_DNS_MESSAGE ((guchar)2)
+/**
+ * EOS_OSTREE_AVAHI_SIZE_LEVEL_FIT_SINGLE_ETHERNET_PACKET:
+ *
+ * TODO
+ */
+#define EOS_OSTREE_AVAHI_SIZE_LEVEL_FIT_SINGLE_ETHERNET_PACKET ((guchar)3)
+/**
+ * EOS_OSTREE_AVAHI_SIZE_LEVEL_FIT_SINGLE_MULTICAST_DNS_PACKET:
+ *
+ * TODO
+ */
+#define EOS_OSTREE_AVAHI_SIZE_LEVEL_FIT_SINGLE_MULTICAST_DNS_PACKET ((guchar)4)
+/**
+ * EOS_OSTREE_AVAHI_SIZE_LEVEL_FIT_16_BIT_LIMIT:
+ *
+ * TODO
+ */
+#define EOS_OSTREE_AVAHI_SIZE_LEVEL_FIT_16_BIT_LIMIT ((guchar)5)
+/**
+ * EOS_OSTREE_AVAHI_SIZE_LEVEL_ABSOLUTELY_LAX:
+ *
+ * TODO
+ */
+#define EOS_OSTREE_AVAHI_SIZE_LEVEL_ABSOLUTELY_LAX ((guchar)6)
+
+gboolean eos_ostree_avahi_service_file_check_options (GVariant  *options,
+                                                      GError   **error);
+gboolean eos_ostree_avahi_service_file_generate (const gchar   *avahi_service_directory,
+                                                 OstreeRepo    *repo,
```

Comment:
OTOH, passing OstreeRepo allows us to extract some more or just different data from the repo for the future versions of TXT records. Passing refs and summary timestamp explicitly can be risky, because we may possibly not use them in some future version of TXT records.

So maybe just stuffing everything to the options variant is not an entirely bad idea in order to avoid both using ostree API and having useless parameters in future.